src/components/IDO/IDO.tsx

```diff
@@ -0,0 +1,203 @@
+import React, { useEffect } from 'react'
+import { Grid, Typography, Box, CardMedia } from '@material-ui/core'
+import useStyles from './style'
+import ExchangeAmountInput from '@components/Inputs/ExchangeAmountInput/ExchangeAmountInput'
+import { printBN } from '@consts/utils'
+import { OutlinedButton } from '@components/OutlinedButton/OutlinedButton'
+import watchIcon from '@static/svg/watch.svg'
+import solanaIcon from '@static/svg/solanaToken.svg'
+import dollarIcon from '@static/svg/dollarToken.svg'
+import invariantIcon from '@static/svg/invariantToken.svg'
+import invariantLogo from '@static/svg/invariantLogo.svg'
+import { SwapToken } from '@components/Swap/Swap'
+
+export interface IIDO {
+  tokens: SwapToken[]
+}
+interface ICurrencyData {
+  bitcoin: ICurrencyObject
+  tether: ICurrencyObject
+  solana: ICurrencyObject
+  ethereum: ICurrencyObject
+}
+interface ICurrencyObject {
+  usd: number
+}
+
+const IDO: React.FC<IIDO> = ({ tokens }) => {
+  const classes = useStyles()
+  const getButtonMessage = () => {
+    return 'Connect a wallet'
+  }
+  const [tokenFromIndex, setTokenFromIndex] = React.useState<number | null>(
+    tokens.length ? 0 : null
+  )
+  const [amountFrom, setAmountFrom] = React.useState<string>('')
+  const [totalDeposited, setTotalDeposited] = React.useState<number>(0)
+  const [totalSolContributed, setTotalSolContributed] = React.useState<number>(0)
+  const [tokenPrice, setTokenPrice] = React.useState<number>(0)
+  const [invariantForSale, setInvariantForSale] = React.useState<number>(0)
+  const [currencyData, setCurrencyData] = React.useState<ICurrencyData>({})
+
+  useEffect(() => {
+    if (tokenFromIndex !== null) {
+      setAmountFrom('0.000000')
+    }
+    setTotalDeposited(45.32)
+    setTotalSolContributed(122124846)
+    setTokenPrice(218.839)
+    setInvariantForSale(20000000)
+  }, [])
+
+  function numberWithSpaces(x: number) {
+    return x.toString().replace(/\B(?=(\d{3})+(?!\d))/g, ' ')
+  }
+  useEffect(() => {
```

Comment:
any connection with api/redux takes place in containers in our project, components should only get necessary values from props